akkaStreamsCustomStage/src/main/scala/com/lightbend/scala/modelServer/modelServer/AkkaModelServer.scala

```diff
@@ -38,48 +44,32 @@ object AkkaModelServer {
     .withGroupId(MODELS_GROUP)
     .withProperty(ConsumerConfig.AUTO_OFFSET_RESET_CONFIG, "earliest")
 
-  def main(args: Array[String]): Unit = {
 
-    val modelStream: Source[ModelWithDescriptor, Consumer.Control] =
-      Consumer.atMostOnceSource(modelConsumerSettings, Subscriptions.topics(MODELS_TOPIC))
-        .map(record => ModelToServe.fromByteArray(record.value())).filter(_.isSuccess).map(_.get)
-        .map(record => ModelWithDescriptor.fromModelToServe(record)).filter(_.isSuccess).map(_.get)
+  def main(args: Array[String]): Unit = {
 
     val dataStream: Source[WineRecord, Consumer.Control] =
       Consumer.atMostOnceSource(dataConsumerSettings, Subscriptions.topics(DATA_TOPIC))
-        .map(record => DataRecord.fromByteArray(record.value())).filter(_.isSuccess).map(_.get)
-
-    val model = new ModelStage()
-
-    def keepModelMaterializedValue[M1, M2, M3](m1: M1, m2: M2, m3: M3): M3 = m3
+        .map(record => DataRecord.fromByteArray(record.value))
+        .collect { case Success(a) => a }
 
-    val modelPredictions: Source[Option[Double], ReadableModelStateStore] = Source.fromGraph(
-      GraphDSL.create(dataStream, modelStream, model)(keepModelMaterializedValue) { implicit builder => (d, m, w) =>
-        import GraphDSL.Implicits._
+    val modelPredictions: Source[Option[Double], ModelStateStore] =
+      dataStream.viaMat(new ModelStage)(Keep.right)
 
-        // wire together the input streams with the model stage (2 in, 1 out)
-        /*
-                            dataStream --> |       |
-                                           | model | -> predictions
-                            modelStream -> |       |
-          */
-
-        d ~> w.dataRecordIn
-        m ~> w.modelRecordIn
-        SourceShape(w.scoringResultOut)
-      }
-    )
-
-    val materializedReadableModelStateStore: ReadableModelStateStore =
+    val modelStateStore: ModelStateStore =
       modelPredictions
-        //        .map(println(_))
         .to(Sink.ignore) // we do not read the results directly
         .run() // we run the stream, materializing the stage's StateStore
```

Comment:
Note that this can now be written as one single "line", but it's good to split it up to see the parts better, but if you really want to have few lines on a slide it can be:

```scala
    val modelStateStore: ModelStateStore =
      Consumer.atMostOnceSource(dataConsumerSettings, Subscriptions.topics(DATA_TOPIC))
        .map(record => DataRecord.fromByteArray(record.value))
        .collect { case Success(a) => a }
        .viaMat(new ModelStage)(Keep.right)
        .to(Sink.ignore) // we do not read the results directly
        .run() // we run the stream, materializing the stage's StateStore
```

cc @deanwampler @jboner

Reply:
Very nice. Thanks, @patriknw. Overall great cleanup.